src/auth/mod.rs

```diff
@@ -26,53 +26,15 @@ pub struct DockerAuthConfig {
 }
 
 /// Get a credential (RegistryAuth) for the given Reference.
-/// First, it will try to find auth info in the local
-/// `auth.json`. If there is not one, it will
-/// ask one from the [`crate::secure_channel::SecureChannel`], which connects
-/// to the GetResource API of Attestation Agent.
-/// Then, it will use the `auth.json` to find
-/// a credential of the given image reference.
-#[cfg(feature = "getresource")]
+/// The path can be from different places. Like `path://` or
+/// `kbs://`.
 pub async fn credential_for_reference(
     reference: &Reference,
-    secure_channel: std::sync::Arc<tokio::sync::Mutex<crate::secure_channel::SecureChannel>>,
     auth_file_path: &str,
 ) -> Result<RegistryAuth> {
-    // if Policy config file does not exist, get if from KBS.
-    if !Path::new(auth_file_path).exists() {
-        secure_channel
-            .lock()
-            .await
-            .get_resource(RESOURCE_DESCRIPTION, HashMap::new(), auth_file_path)
-            .await?;
-    }
+    let auth = crate::resource::get_resource(auth_file_path).await?;
```

Comment:
**BUG**: Now `auth_file_path` is `/run/image-security/auth.json`, not a KBS resource URI.

Reply:
Thanks for pointing out. Fixed